# Bayes Theorem Example

Suppose a company produces light bulbs and has two factories, A and B. Factory A produces 60% of the bulbs, while factory B produces the remaining 40%. Of the bulbs produced by factory A, 95% are of good quality, while only 80% of the bulbs produced by factory B are of good quality. A customer buys a bulb at random and finds that it is of good quality. What is the probability that the bulb came from factory A?

To solve this problem using Bayes' theorem, we can define the following probabilities:

* P(A) = probability that the bulb came from factory A (0.6)
* P(B) = probability that the bulb came from factory B (0.4)
* P(G|A) = probability that the bulb is of good quality given that it came from factory A (0.95)
* P(G|B) = probability that the bulb is of good quality given that it came from factory B (0.80)
* P(G) = probability that the bulb is of good quality (unknown)

Using Bayes' theorem, we can calculate the probability that the bulb came from factory A given that it is of good quality:

P(A|G) = P(G|A) * P(A) / (P(G|A) * P(A) + P(G|B) * P(B))

In [1]:
(0.95 * 0.6) / ((0.95 * 0.6) + (0.8 * 0.4))

0.6404494382022471

Therefore, the probability that the bulb came from factory A given that it is of good quality is approximately 0.64, or 64%. This means that it is more likely that the bulb came from factory A than from factory B.